### PR TITLE
test: playwright integration

### PR DESCRIPTION
**Pull request by malviyagaurav**

### CODE REVIEW

File: apps/desktop/.github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Install Playwright Browsers
+        run: pnpm exec playwright install --with-deps
+      - name: Run Playwright tests
+        run: pnpm exec playwright test
+      - uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: playwright-report
+          path: playwright-report/
+          retention-days: 30
```
Comment (irshadCypherock) [inline apps/desktop/.github/workflows/playwright.yml]: This should be in the project root. Also, integrate it with the already existing `test` workflow.

File: apps/desktop/e2e/base.spec.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  Page,
+  _electron as electron,
+} from '@playwright/test';
+import { findLatestBuild, parseElectronApp } from 'electron-playwright-helpers';
+
+let electronApp: ElectronApplication;
+
+test.beforeAll(async () => {
+  // find the latest build in the out directory
+  const latestBuild = findLatestBuild('release/1.0.0');
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Don't hard code the version here.

File: apps/desktop/e2e/base.spec.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    console.log(`Window opened: ${filename}`);
+
+    // capture errors
+    page.on('pageerror', error => {
+      console.error(error);
+    });
+    // capture console messages
+    page.on('console', msg => {
+      console.log(msg.text());
+    });
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Don't log console messages, it'll bloat the CI

File: apps/desktop/e2e/base.spec.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    executablePath: appInfo.executable,
+  });
+  electronApp.on('window', async page => {
+    const filename = page.url().split('/').pop();
+    console.log(`Window opened: ${filename}`);
+
+    // capture errors
+    page.on('pageerror', error => {
+      console.error(error);
+    });
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: I think we should treat this as an error

File: apps/desktop/e2e/base.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+test.beforeEach(async () => {
+  const splash = await electronApp.firstWindow();
+  await splash.waitForEvent('close');
+  screen = await electronApp.firstWindow();
+});
+
+test('record', async () => {
+  await screen.pause();
+});
+
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Remove record test

File: apps/desktop/e2e/base.spec.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  const newwindow = screen.getByRole('heading', {
+    name: 'Your X1 Vault is successfully authenticated',
+  });
+  await newwindow.waitFor();
+  expect(newwindow).toBeVisible();
+});
+
+// test('Device auth fail due to internet disconnection', async () => {
+//
+// })
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Remove commented code

File: apps/desktop/e2e/base.spec.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
+/**
+ * Example Playwright script for Electron
+ * showing/testing various API features
+ * in both renderer and main processes
+ */
+
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Remove comment

File: apps/desktop/e2e/base.spec.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  expect(newwindow).toBeVisible();
+});
+
+test('Device authentication successful', async () => {
+  const newwindow = screen.getByRole('heading', {
+    name: 'Your X1 Vault is successfully authenticated',
+  });
+  await newwindow.waitFor();
+  expect(newwindow).toBeVisible();
+});
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: In `e2e` only add tests which can be run on CI.

File: apps/desktop/e2e/base.spec.ts [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+//
+// })
+
+test('Device disconnection during device auth', async () => {
+  await screen
+    .getByRole('heading', {
+      name: 'Your X1 Vault will now be authenticated through Cypherock to check its authenticity...(?)',
+    })
+    .waitFor({ timeout: 300000 });
+  const newwindow = screen.getByRole('heading', {
```
Comment (irshadCypherock) [inline apps/desktop/e2e/base.spec.ts]: Use pascal case for variables. Example `newWindow`.

File: apps/desktop/package.json [hunk 10]
```diff
@@ -45,6 +45,7 @@
     "@cypherock/tsconfig": "workspace:^",
     "@electron/notarize": "^1.2.3",
     "@jest/globals": "^29.4.1",
+    "@playwright/test": "^1.34.3",
```
Comment (irshadCypherock) [inline apps/desktop/package.json:49]: Add a script in package.json to run `e2e` tests

File: apps/desktop/playwright.config.ts [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    //   use: { ..devices['Desktop Chrome'], channel: 'chrome' },
+    // },
+  ],
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
+  //   command: 'npm run start',
+  //   url: 'http://127.0.0.1:3000',
+  //   reuseExistingServer: !process.env.CI,
+  // },
```
Comment (irshadCypherock) [inline apps/desktop/playwright.config.ts]: Remove commented code

File: apps/desktop/tsconfig.json [hunk 12]
```diff
@@ -4,6 +4,6 @@
     "lib": ["DOM", "DOM.Iterable", "ESNext"]
   },
   "extends": "@cypherock/tsconfig/browser.json",
-  "include": ["renderer"],
+  "include": ["renderer", "playwright.config.ts"],
```
Comment (irshadCypherock) [inline apps/desktop/tsconfig.json]: I don't think we need this here.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: apps/desktop/e2e/base.spec.ts (hunk 3 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: Removed function

File: apps/desktop/e2e/base.spec.ts (hunk 4 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: Removed function

File: apps/desktop/e2e/base.spec.ts (hunk 5 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: Skipped in the code so that it will not run in CI

File: apps/desktop/e2e/base.spec.ts (hunk 8 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: Added one separate folder for device dependent tests named me2e.

File: apps/desktop/e2e/base.spec.ts (hunk 9 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: changed to camelCase

File: apps/desktop/playwright.config.ts (hunk 11 above)
Comment (malviyagaurav) [reply apps/desktop/playwright.config.ts]: Removed unnecessary comments

File: apps/desktop/tsconfig.json (hunk 12 above)
Comment (malviyagaurav) [reply apps/desktop/tsconfig.json]: done

File: apps/desktop/package.json (hunk 10 above)
Comment (malviyagaurav) [reply apps/desktop/package.json:49]: e2e has been added in script

File: apps/desktop/e2e/base.spec.ts (hunk 7 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: Removed

File: apps/desktop/e2e/base.spec.ts (hunk 6 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: removed

File: apps/desktop/e2e/base.spec.ts (hunk 2 above)
Comment (malviyagaurav) [reply apps/desktop/e2e/base.spec.ts]: fixed

File: apps/desktop/.github/workflows/playwright.yml (hunk 1 above)
Comment (malviyagaurav) [reply apps/desktop/.github/workflows/playwright.yml]: changes made

**Pull request closed without merging**